src/object/cli_obj.c

```diff
@@ -1074,6 +1105,9 @@ obj_comp_cb(tse_task_t *task, void *data)
 	if (obj_retry_error(task->dt_result) || obj_auxi->io_retry)
 		io_retry = true;
 
+	if (io_retry && task->dt_result == -DER_INPROGRESS)
```

Comment:
I understand that the shard update child task can get -DER_INPROGRESS as L989 above, not sure if the parent update IO task can get -DER_INPROGRESS?

Reply:
The L989 will set the "retry_with_leader" as true when detect the "DER_INPROGRESS", that seems enough for the cases of update/punch hit non-committed DTX.

Reply:
and also for fetch against 'prepared' dtx.